code/weapon/weapon.h

```diff
@@ -176,6 +178,42 @@ typedef struct beam_weapon_section_info {
 	generic_anim texture;					// texture anim/bitmap
 } beam_weapon_section_info;
 
+enum class Type5BeamRotAxis {
+	STARTPOS_OFFSET,
+	ENDPOS_OFFSET, 
+	STARTPOS_NO_OFFSET,
+	ENDPOS_NO_OFFSET,
+	CENTER,
+	UNSPECIFIED
+};
+
+enum class Type5BeamPos {
+	RANDOM_OUTSIDE,
+	RANDOM_INSIDE,
+	CENTER,
+	SAME_RANDOM
+};
+
+typedef struct type5_beam_info {
+	bool no_translate;                           // true if the end pos parameters were left unspecified
+	Type5BeamPos start_pos;             // whether it starts from the center or a type 0 or 1 beam kind of random
+	Type5BeamPos end_pos;               // same as above but but with an extra 'same random as start' option
+	vec3d start_pos_offset;                      // position simply added to start pos (possibly manipulated by the bools below)
+	vec3d end_pos_offset;                        // position simply added to end pos (possibly manipulated by the bools below)
+	vec3d start_pos_rand;                        // same as above but a randomly chosen between defined value for each axis and its negative
+	vec3d end_pos_rand;
+	bool target_orient_positions;                // if true, offsets are oriented relative to the target, else the shooter's pov
+	bool target_scale_positions;                 // if true, offsets are scaled by target radius, else its a fixed span from the shooters pov
+	                                             // regardless of distance
+	float continuous_rot;                        // radians per sec rotation over beam lifetime
+	Type5BeamRotAxis continuous_rot_axis;   // axis to do ^^ on
```

Comment:
Pedantic request, please change

Reply:
Err, change what? Be more specific with the comment, you mean?